docs/photonics/examples/depletion_waveguide.py

```diff
@@ -0,0 +1,135 @@
+
+# # +
+xpn = 0
+NA = 1e17
+ND = 1e17
+V = 0
+wavelength = 1.55
+
+
+def define_index(mesh, V, xpn, NA, ND, wavelength):
+    basis0 = Basis(mesh, ElementTriP0())
+    epsilon = basis0.zeros(dtype=complex)
+    for subdomain, n in {"core": 3.45, "slab": 3.45}.items():
+        epsilon[basis0.get_dofs(elements=subdomain)] = n**2
+    epsilon += basis0.project(
+        lambda x: index_pn_junction(x[0], xpn, NA, ND, V, wavelength) ** 2,
+        dtype=complex,
+    )
+    for subdomain, n in {"clad": 1.444}.items():
+        epsilon[basis0.get_dofs(elements=subdomain)] = n**2
+    return basis0, epsilon
+
+
+basis0, epsilon = define_index(mesh, V, xpn, NA, ND, wavelength)
+basis0.plot(epsilon.real, colorbar=True).show()
+basis0.plot(epsilon.imag, colorbar=True).show()
+
+# -
+# The index change is weak compared to the contrast between silicon and silicon dioxide, but it is accompanied by a change in absorption which is easier to observe. As voltage is increased, the region without charge widens, which is the mechanism behind depletion modulation:
+
+V = -4
+basis0, epsilon = define_index(mesh, V, xpn, NA, ND, wavelength)
+basis0.plot(epsilon.imag, colorbar=True).show()
+# -
+
+# And now we can mode solve as before, and observe the change in effective index and absorption of the fundamental mode as a function of applied voltage for given junction position and doping levels:
+
+# +
+voltages = [0, -1, -2, -3, -4]
+neff_vs_V = []
+for V in voltages:
+    basis0, epsilon = define_index(mesh, V, xpn, NA, ND, wavelength)
+    lams, basis, xs = compute_modes(basis0, epsilon, wavelength=wavelength, num_modes=1, order=2)
+    neff_vs_V.append(lams)
+
+plt.plot(voltages, neff_vs_V.real)
+plt.title(f"NA = {NA}, ND = {ND}, xpn = {xpn}, wavelength = {wavelength}")
+plt.xlabel("Voltage (V)")
+plt.ylabel("neff0")
+
+plt.plot(voltages, k_to_alpha_dB(neff_vs_V.imag, wavelength))
+plt.title(f"NA = {NA}, ND = {ND}, xpn = {xpn}, wavelength = {wavelength}")
+plt.xlabel("Voltage (V)")
+plt.ylabel("absorption (dB/cm)")
+
+# -
+
+# References:
+
+# From Chrostowski, L., & Hochberg, M. (2015). Silicon Photonics Design: From Devices to Systems. Cambridge University Press. doi: 10.1017/CBO9781316084168
+#     Citing:
+#     (1) R. Soref and B. Bennett, "Electrooptical effects in silicon," in IEEE Journal of Quantum Electronics, vol. 23, no. 1, pp. 123-129, January 1987, doi: 10.1109/JQE.1987.1073206.
+#     (2) Reed, G. T., Mashanovich, G., Gardes, F. Y., & Thomson, D. J. (2010). Silicon optical modulators. Nature Photonics, 4(8), 518–526. doi: 10.1038/nphoton.2010.179
+#     (3) M. Nedeljkovic, R. Soref and G. Z. Mashanovich, "Free-Carrier Electrorefraction and Electroabsorption Modulation Predictions for Silicon Over the 1–14- $\mu\hbox{m}$ Infrared Wavelength Range," in IEEE Photonics Journal, vol. 3, no. 6, pp. 1171-1180, Dec. 2011, doi: 10.1109/JPHOT.2011.2171930.
```

Comment:
See https://github.com/HelgeGehring/femwell/blob/main/docs/photonics/examples/bragg_filter.py , if you mention the references in the text, you can automatically generate the references-list :)

Reply:
Yes I already updated the bibliography, let me cite properly